src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -172,16 +241,20 @@ private static void HookOrProcessResult(HttpWebRequest request)
             asyncCallbackModifier(readAsyncContext, callback.AsyncCallback);
         }
 
-        private static void ProcessResult(IAsyncResult asyncResult, AsyncCallback asyncCallback, HttpWebRequest request, Activity activity, object result, bool forceResponseCopy)
+        private static void ProcessResult(IAsyncResult asyncResult, AsyncCallback asyncCallback, Activity activity, object result, bool forceResponseCopy)
         {
             // We could be executing on a different thread now so set the activity.
-            Activity.Current = activity;
+            Debug.Assert(Activity.Current == null || Activity.Current == activity, "There was an unexpected active Activity on the result thread.");
+            if (Activity.Current == null)
+            {
+                Activity.Current = activity;
```

Comment:
will `activity` be ever null here?

Reply:
No. If Activity isn't created due to no listeners or a decision not to sample, we don't hook the request and `ProcessResult` is never called.